feat: add react-native code to guide - OKTA-233817 (#1093)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/handle-authentication-responses/react-native/handle-responses.md

### Response handling

> The SDK provides two options, Promise and Event listener, to handle responses. You can choose either approach base on your preference.

Reviewer: ```suggestion
The SDK provides two options, Promise and Event listener, to handle responses. You can choose either approach based on your preference.
```

#### Handle response in Promise

> The `signIn` function will return a Promise if user credentials (username, password) had been provided. Success and error reposonse can be easily handled in standard Promise way.

Reviewer: ```suggestion
The `signIn` function will return a Promise if user credentials (username, password) have been provided. Success and error responses can be easily handled in standard Promise way.
```

> The `signIn` function will return a Promise if user credentials (username, password) have been provided. Success and error responses can be easily handled in standard Promise way.

Reviewer 2: bolded my edits for these two sentences:

The `signIn` function **returns** a Promise if user credentials (username, password) **are** provided. Success and error responses can be easily handled in **a** standard Promise way.

Reviewer 3: will change to `the standard Promise way` instead of `a standard Promise way`

```javascript
import { signIn } from '@okta/okta-react-native';

class LoginScreen extends React.Component {

    /* code from last section */

    login() {
        const { username, password } = this.state;
        signIn({ username, password })
            .then(token => {
                // handle success response
            })
            .catch(error => {
                // handle error response
            });
    }

    /* code from last section */
}
```

#### Handle response in Event listener

> You can also register event listeners from `EventEmitter` function in the SDK to handle responses. 

Reviewer: ```suggestion
You can also register event listeners from the `EventEmitter` function in the SDK to handle responses. 
```

> **Note:** Please make sure listeners will be properly unregistered in component.

Reviewer: "in the component"? Not sure how to fix this, but something is missing.

> **Note:** Please make sure listeners will be properly unregistered when component will unmount.

Author: There is a lifecycle method in react called `componentWillMount`, those listeners should be unregistered there. I think the `will unmount` should still be kept.

> **Note:** Please make sure listeners will be properly unregistered when the component will unmount.

Reviewer 2: we avoid the use of "please" in content, and avoid future tense. Bolded the chagnes:

**Note**: Make sure **that** listeners **are** properly unregistered when the component **unmounts**.

Reviewer 3: Thanks for the comment! I still want to keep the `will unmount` part due to https://github.com/okta/okta-developer-docs/pull/1093#discussion_r416918598

```javascript
import { EventEmitter } from '@okta/okta-react-native';

class LoginScreen extends React.Component {

    /* code from last section */

    // register listeners
    componentDidMount() {
        EventEmitter.addListener('signInSuccess', (e) => {
            // handle success response
        });

        EventEmitter.addListener('onError', (e) => {
            // handle error response
        });
    }

    // unregister listeners
    componentWillUnmount() {
        EventEmitter.removeAllListeners('signInSuccess');
        EventEmitter.removeAllListeners('onError');
    }

    /* code from last section */
}
```

#### Handle transaction state

> As transaction states are evaluated during primary authentication, this SDK also exposes a `getAuthClient` function to return an instance of `@okta/okta-auth-js` client to handle [Authentication API](/docs/reference/api/authn/) communication. Please refer to [Node JS and React Native Usage](https://github.com/okta/okta-auth-js#node-js-and-react-native-usage) for more information.

Reviewer 2: second sentence edit (change in bold):

**See** [Node JS and React Native Usage](https://github.com/okta/okta-auth-js#node-js-and-react-native-usage) for more information.

> As `transaction states` are evaludated during primary authentication, this SDK also exposes `getAuthClient` function to return an instance of `@okta/okta-auth-js` client to handle [Authentication API](https://developer.okta.com/docs/reference/api/authn/) communication. Please refer to [Node JS and React Native Usage](https://github.com/okta/okta-auth-js#node-js-and-react-native-usage) for more information.

Reviewer: Is `transaction states` an actual value in the code? If it isn't, then it shouldn't be code formatted.

> As `transaction states` are evaludated during primary authentication, this SDK also exposes `getAuthClient` function to return an instance of `@okta/okta-auth-js` client to handle [Authentication API](https://developer.okta.com/docs/reference/api/authn/) communication. Please refer to [Node JS and React Native Usage](https://github.com/okta/okta-auth-js#node-js-and-react-native-usage) for more information.

Reviewer: ```suggestion
As `transaction states` are evaluated during primary authentication, this SDK also exposes a `getAuthClient` function to return an instance of `@okta/okta-auth-js` client to handle [Authentication API](/docs/reference/api/authn/) communication. Please refer to [Node JS and React Native Usage](https://github.com/okta/okta-auth-js#node-js-and-react-native-usage) for more information.
```

> One general use case is you can resume transaction from the `authClient` to continue handling transactions for status other than `SUCCESS`.

Reviewer: ```suggestion
One general use case is that you can resume a transaction from the `authClient` to continue handling transactions for statuses other than `SUCCESS`.
```
